backend/app/Repositories/EloquentButtonDataPageViews.php

```diff
@@ -18,4 +19,31 @@ public function countBetweenDate(DatePeriod $filterData, int $websiteId): int
             ->whereBetween('visit_time', [$filterData->getStartDate(), $filterData->getEndDate()])
             ->count();
     }
+
+    public function getAvgTimeOnPageBetweenDate(DatePeriod $filterData, int $websiteId): int
+    {
+        $bindings = [
+            $websiteId,
+            $filterData->getStartDate(),
+            $filterData->getEndDate()
+        ];
+
+        $sql = 'SELECT AVG(grouped.avg_by_session) AS avg_time
```

Comment:
Maybe, you should divide your  query into smaller methods

Reply:
Done )